Flux switch: avoid updates when off

amelchio opened the pull request:
## Description:

Calling turn_on when the switch is already on would orphan the existing
time tracker, losing our ability to cancel it when turn_off is called.

**Related issue (if applicable):** fixes #6958

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

amelchio committed:
Flux switch: avoid updates when off

Calling turn_on when the switch is already on would orphan the existing
time tracker, losing our ability to cancel it when turn_off is called.

balloob commented on homeassistant/components/switch/flux.py:
```diff
@@ -140,8 +140,11 @@ def turn_on(self, **kwargs):
-        self.unsub_tracker = track_time_change(self.hass, self.flux_update,
-                                               second=[0, 30])
+
+        if self.unsub_tracker is None:
```
I would expect instead of doing this check to just add an `if self._state: return` to the beginning.

I also guess that `self._state` is not necessary, it can be dereived from `self.unsub_tracker is not None`

amelchio replied:
Right, I didn't consider cleanups. Added those suggestions now.

amelchio committed:
Cleanups

The self.is_on property can now be found from self.unsub_tracker, so
get rid of the self._state attribute.

Add an entry guard to turn_on, making further conditionals unnecessary.

balloob commented:
Woop thanks 🐬
